okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpEngine.java

```diff
@@ -899,7 +899,15 @@ public void readResponse() throws IOException {
         bufferedRequestBody.close();
       }
 
-      return readNetworkResponse();
+      Response response = readNetworkResponse();
+
+      int code = response.code();
+      if ((code == 204 || code == 205) && response.body().contentLength() > 0) {
```

Comment:
I wanted to put this closer to the parsing of this information, but since `HttpUrlConnection` specifically allows this behavior I opted for here. Open to suggestions if there's a more appropriate place.